src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -44,6 +50,80 @@ public static Index parseIndex(String oneBasedIndex) throws ParseException {
         return Index.fromOneBased(Integer.parseInt(trimmedIndex));
     }
 
+    /**
+     * Parses a range in the format "a-b" into a list of {@code Index} objects.
+     *
+     * @param range The range string to parse.
+     * @return A list of {@code Index} objects from start to end of the range.
+     * @throws ParseException If the format is invalid or bounds are incorrect.
+     */
+    public static List<Index> parseInterval(String range) throws ParseException {
+        String[] bounds = range.split("-");
+        System.out.println(bounds.toString());
+        // Validate that the range contains exactly two parts
+        if (bounds.length != 2) {
+            throw new ParseException(MESSAGE_INVALID_INTERVAL_FORMAT);
+        }
+
+        // Parse start and end indices
+        Index startIndex = parseIndex(bounds[0]);
+        Index endIndex = parseIndex(bounds[1]);
+
+        // Validate that startIndex is less than or equal to endIndex
+        if (startIndex.getOneBased() > endIndex.getOneBased()) {
+            throw new ParseException(MESSAGE_INVALID_INTERVAL);
+        }
+
+        // Generate and return the list of indices from start to end
+        return IntStream.rangeClosed(startIndex.getOneBased(), endIndex.getOneBased())
+                .mapToObj(Index::fromOneBased)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Parses a string containing indices and ranges (e.g., "1,2,3,5-9") into a list of {@code Index} objects.
+     * Accepts both single indices and ranges separated by spaces.
+     *
+     * @param args The string containing indices and/or ranges to parse.
+     * @return A list of {@code Index} objects representing the parsed indices and ranges.
+     * @throws ParseException If any part of the input is not a valid index or range.
+     */
+    public static List<Index> parseIndices(String args) throws ParseException {
+        requireNonNull(args);
+        String trimmedIndices = args.trim();
+
+        try {
+            return Arrays.stream(trimmedIndices.split(","))
+                    .flatMap(arg -> parseArgAsStream(arg))
+                    .collect(Collectors.toList());
+        } catch (RuntimeException e) {
+            String causeMessage = e.getCause().getMessage();
```

Comment:
This is violating LoD. But I couldn't really find a nice way to deal with this. Do let me know if you have any ideas

Reply:
I think this doesn't matter anymore. Just asked prof:
```
Dear Zhanzhi, 
We don't tP grade code quality for violating principles such as LoD.
-damith
```